service/controller/app/v1/resource/chartoperator/resource.go

```diff
@@ -186,11 +191,60 @@ func (r Resource) installChartOperator(ctx context.Context, cr v1alpha1.App, hel
 	}
 
 	{
-		err = helmClient.InstallReleaseFromTarball(ctx, tarballPath, namespace, helm.ReleaseName(release), helm.ValueOverrides(chartOperatorValue))
+		err = cc.HelmClient.InstallReleaseFromTarball(ctx, tarballPath, namespace, helm.ReleaseName(release), helm.ValueOverrides(chartOperatorValue))
 		if err != nil {
 			return microerror.Mask(err)
 		}
 	}
 
+	{
+		// We wait for the chart-operator deployment to be ready so the
+		// chart CRD is installed. This allows the chart
+		// resource to create CRs in the same reconcilation loop.
+		r.logger.LogCtx(ctx, "level", "debug", "message", "waiting for ready chart-operator deployment")
+
+		o := func() error {
+			err := r.checkDeploymentReady(ctx, cc.K8sClient)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			return nil
+		}
+
+		// Wait for chart-operator to be deployed. If it takes longer than
+		// the timeout the chartconfig CRs will be created during the next
+		// reconciliation loop.
+		b := backoff.NewConstant(30*time.Second, 5*time.Second)
```

Comment:
30 seconds for 5 minutes is okay?

Reply:
30 secs is the total wait time. The interval is to check every 5 secs.